### PR TITLE
Expand micrometer test coverage #4582

### PR DESCRIPTION
**Pull request by JiriOndrusek**
fixes https://github.com/apache/camel-quarkus/issues/4582

1. I added a test for jmx access to metrics (JVM only). For that porpose I had to add a dependency `io.micrometer:micrometer-registry-jmx` into CQ BOM.
2. Prometheus metrics ignores negative values for counter. I put that information into `limitations.adoc`, because there is Camel header `decrement` which is meant to be used with counter metrics. Possibly better place for such information is Camel itself.

### CODE REVIEW

**Comment by ppalaga** [comment]
> So maybe a better way would be to add `io.quarkiverse.micrometer.registry:quarkus-micrometer-registry-jmx` only to the test BOM.

I see it managed in quarkus-bom: https://repo1.maven.org/maven2/io/quarkus/quarkus-bom/3.1.0.Final/quarkus-bom-3.1.0.Final.pom

> It's a bit of an open question whether we even need to test the JMX bits at all, given it's already tested elsewhere.

JMX support is underway for "GraalVM for JDK 17 / JDK 20 (June 13, 2023)", so these tests may come in handy for native testing, no?

**Comment by jamesnetherton** [comment]
> I see it managed in quarkus-bom

That's just the plain `micrometer-registry-jmx` dependency. Not the Quarkus extension that lives here:

https://github.com/quarkiverse/quarkus-micrometer-registry/

**Comment by ppalaga** [comment]
> > I see it managed in quarkus-bom
> 
> That's just the plain `micrometer-registry-jmx` dependency. Not the Quarkus extension that lives here:
> 
> https://github.com/quarkiverse/quarkus-micrometer-registry/

I am blind, sorry!

**Comment by JiriOndrusek** [comment]
> I am not sure if we want to be supporting `micrometer-registry-jmx` directly in this project. Otherwise, we have to maintain the dependencies, tests and docs explaining how to configure it.
> 
> Support for the different Micrometer registry implementations lives here:
> 
> https://github.com/quarkiverse/quarkus-micrometer-registry/
> 
> So maybe a better way would be to add `io.quarkiverse.micrometer.registry:quarkus-micrometer-registry-jmx` only to the test BOM. And just replace the table in our [usage docs](https://camel.apache.org/camel-quarkus/next/reference/extensions/micrometer.html#extensions-micrometer-usage) with a link to [here](https://quarkiverse.github.io/quarkiverse-docs/quarkus-micrometer-registry/dev/index.html) .
> 
> It's a bit of an open question whether we even need to test the JMX bits at all, given it's already tested elsewhere.
> 
> WDYT @ppalaga?

The dependency is changed and moved.
Table in the doc is replaced but I had to keep onel dependency for prometheus which is not part of the quarkiverse doc.

**Review by ppalaga — APPROVED** [review]

**Comment by ppalaga** [comment]
Thanks @JiriOndrusek!

File: extensions/micrometer/runtime/src/main/doc/limitations.adoc [hunk 1]
```diff
@@ -0,0 +1,8 @@
+=== Exposing Micrometer statisctis in Jmx
+
+The `Exposing Micrometer statisctis in Jmx` action is not available in native mode as JMX is not supported on GraalVM.
```
Comment (jamesnetherton) [inline extensions/micrometer/runtime/src/main/doc/limitations.adoc:3]:
```suggestion
The `Exposing Micrometer statistics in JMX` action is not available in native mode as JMX is not supported on GraalVM.
```

Comment (JiriOndrusek) [reply extensions/micrometer/runtime/src/main/doc/limitations.adoc]: fixed

**Pull request merged**